src/python/grpcio/grpc/_channel.py

```diff
@@ -374,13 +376,31 @@ def details(self):
                 self._state.condition.wait()
             return _common.decode(self._state.details)
 
+    def debug_error_string(self):
+        with self._state.condition:
+            while self._state.debug_error_string is None:
+                self._state.condition.wait()
+            return _common.decode(self._state.debug_error_string)
+
+    def _format_debug_error_string(self):
+      return self._state.debug_error_string
+
+    def _format_completed_repr(self):
```

Comment:
Insufficient use of `self` to justify living at instance-scope. Maybe pull it out into a module-scope helper? Or... maybe I was wrong before and this can live inside `_repr` after all.

Reply:
Done